examples/community/lpw_stable_diffusion_xl.py

```diff
@@ -1328,18 +1327,7 @@ def _get_add_time_ids(self, original_size, crops_coords_top_left, target_size, d
 
     # Copied from diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion_upscale.StableDiffusionUpscalePipeline.upcast_vae
     def upcast_vae(self):
-        dtype = self.vae.dtype
-        self.vae.to(dtype=torch.float32)
-        use_torch_2_0_or_xformers = isinstance(
-            self.vae.decoder.mid_block.attentions[0].processor,
-            (AttnProcessor2_0, XFormersAttnProcessor),
-        )
-        # if xformers or torch_2_0 is used attention block does not need
-        # to be in float32 which can save lots of memory
-        if use_torch_2_0_or_xformers:
-            self.vae.post_quant_conv.to(dtype)
-            self.vae.decoder.conv_in.to(dtype)
-            self.vae.decoder.mid_block.to(dtype)
+        deprecate("`upcast_vae` is deprecated")
```

Comment:
We should also guide the users about what they should do instead of using `upcast_vae()`. We should also set the version number of thel lib after which this should be removed.

Reply:
Hmm I assumed that no one would be using this method standalone. But looks like there are a few instances where this method is called directly.
https://github.com/search?q=%22pipe.upcast_vae()%22+language:Python&type=code 

Will update the deprecation message and handle the casting inside the method itself for backwards compatiblity.